### PR TITLE
DMP-140 Flyway db migration setup

### PR DESCRIPTION
**Pull request by davet1985**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-140

### Change description ###

Adding flyway database migrations, with a single table to start. Once it's proved that this works across the environments I will create flyway scripts for the other tables.

**Does this PR introduce a breaking change?** (check one with "x")

```
[X] Yes
[ ] No
```

### CODE REVIEW

File: charts/darts-api/values.stg.template.yaml [hunk 1]
```diff
@@ -2,3 +2,17 @@
 java:
   image: ${IMAGE_NAME}
   ingressHost: ${SERVICE_FQDN}
+  keyVaults:
+    darts:
+      secrets:
+        - name: GovukNotifyTestApiKey
+          alias: GOVUK_NOTIFY_API_KEY
+  postgresql:
+    enabled: true
```
Comment (cmagowan-hmcts) [inline charts/darts-api/values.stg.template.yaml:11]: can this be in values.yaml?

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -34,26 +34,25 @@ spring:
...
             token-uri: https://hmctsdartsb2csbox.b2clogin.com/hmctsdartsb2csbox.onmicrosoft.com/B2C_1_darts_externaluser_signin/oauth2/v2.0/token
             jwk-set-uri: https://hmctsdartsb2csbox.b2clogin.com/hmctsdartsb2csbox.onmicrosoft.com/B2C_1_darts_externaluser_signin/discovery/v2.0/keys
 
-
   datasource:
     driver-class-name: org.postgresql.Driver
     url: jdbc:postgresql://${DARTS_API_DB_HOST:localhost}:5432/${DARTS_API_DB_NAME:darts}${DARTS_API_DB_OPTIONS:}
-    username: ${DARTS_API_DB_USERNAME:darts}
-    password: ${DARTS_API_DB_PASSWORD:darts}
+    username: ${DARTS_API_DB_USERNAME:postgres}
```
Comment (cmagowan-hmcts) [inline src/main/resources/application.yaml:48]: having this wrong was useful to prove that environment variables were not being imported correctly as we could see it defaulting to darts, but suppose that shouldn't happen again :-\

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -71,6 +70,30 @@ spring:
...
+  flyway.locations: classpath:db/migration
+#  datasource:
+#    driver-class-name: org.postgresql.Driver
+#    url: jdbc:postgresql://${DB_HOST}:${DB_PORT}/${DB_NAME}${DB_OPTIONS:}
+#    username: ${DB_USER_NAME}
+#    password: ${DB_PASSWORD}
+#    properties:
+#      charSet: UTF-8
+#    hikari:
+#      minimumIdle: 2
```
Comment (cmagowan-hmcts) [inline src/main/resources/application.yaml:83]: should we have all the commented out lines?

File: charts/darts-api/values.stg.template.yaml [hunk 4]
```diff
@@ -2,3 +2,17 @@
 java:
   image: ${IMAGE_NAME}
   ingressHost: ${SERVICE_FQDN}
+  keyVaults:
+    darts:
+      secrets:
+        - name: GovukNotifyTestApiKey
```
[the remark this answers is not in the document]
Comment (davet1985) [reply charts/darts-api/values.stg.template.yaml:8]: @tonymort just picked this up as well, I'll be removing it.

**Comment by cmagowan-hmcts** [comment]
PR description hasn't been updated

File: charts/darts-api/values.stg.template.yaml (hunk 1 above)
Comment (davet1985) [reply charts/darts-api/values.stg.template.yaml:11]: No as it's creating a postgres pod for the staging pipeline deployment, which we don't want in the main deployment.

File: charts/darts-api/values.stg.template.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (davet1985) [reply charts/darts-api/values.stg.template.yaml]: Removed.

File: src/main/resources/application.yaml (hunk 3 above)
Comment (davet1985) [reply src/main/resources/application.yaml]: Removed.

File: src/main/resources/application.yaml (hunk 2 above)
Comment (davet1985) [reply src/main/resources/application.yaml]: Changed back.

**Review by cmagowan-hmcts — APPROVED** [review]

**Pull request merged**